src/errors.rs

```diff
@@ -99,11 +101,35 @@ pub enum OciDistributionError {
     #[error("Failed to convert Config into ConfigFile: {0}")]
     /// Transparent wrapper around `std::string::FromUtf8Error`
     ConfigConversionError(String),
+    /// Verification of the content digest failed
+    #[error("Failed to verify content digest. {0}")]
+    ContentDigestVerificationError(ContentDigestVerificationError),
```

Comment:
Why going this way instead of doing something like:

```rust
#[error("Failed to verify content digest. Expected {expected:?}, got {actual:?}")]
ContentDigestVerificationError{
  expected: String,
  actual: String,
},
```

The other changes to this file can then be dropped.

Reply:
I actually had it this way first, then thought I was going to need to reuse the struct elsewhere and didn't end up using it. I'll put it back and/or refactor when I handle the other digest errors